srcalc/src/main/java/gov/va/med/srcalc/web/controller/admin/EditRuleController.java

```diff
@@ -0,0 +1,163 @@
+    }
+    
+    /**
+     * Creates an {@link EditExistingRule} instance for the identified
+     * rule.
+     * @param ruleId the id of the rule to edit
+     * @return the EditExistingRule instance
+     * @throws InvalidIdentifierException if no such rule exists
+     */
+    @ModelAttribute(NewRuleController.ATTRIBUTE_RULE)
+    private EditExistingRule createEditRuleFromRule(
+            @PathVariable("ruleId") final int ruleId) throws InvalidIdentifierException
+    {
+        fLogger.trace("Creating EditRule from previous Rule with ID: {}.", ruleId);
+        final Rule rule = fAdminService.getRuleById(ruleId);
+        return EditRuleFactory.getInstance(rule, fAdminService);
+    }
+    
+    @RequestMapping(method = RequestMethod.GET)
+    public ModelAndView displayEditRuleForm(
+            @ModelAttribute(NewRuleController.ATTRIBUTE_RULE) final EditExistingRule editRule,
+            @PathVariable("ruleId") final int ruleId)
+                    throws InvalidIdentifierException
+    {
+        final ModelAndView mav = new ModelAndView(Views.EDIT_RULE);
+        mav.addObject("SAVE_URL", ruleId);
+        mav.addObject("isNewRule", false);
+        NewRuleController.addVariablesToModel(mav, editRule, fAdminService);
+        return mav;
+    }
+    
+    @RequestMapping(method = RequestMethod.POST)
+    public ModelAndView handlePost(
+            @ModelAttribute(NewRuleController.ATTRIBUTE_RULE) final EditExistingRule editRule,
+            final BindingResult bindingResult,
+            @PathVariable("ruleId") final int ruleId,
+            @RequestParam(value = "submitButton") final String submitString) throws InvalidIdentifierException
+    {
+        
+        // Test if a remove button was clicked, and remove the proper ValueMatcherBuilder if needed
+        if(submitString.startsWith("remove"))
+        {
+            // Parse the integer that is after the "remove" prefix
+            return removeMatcher(editRule, ruleId, Integer.parseInt(submitString.substring(6)));
+        }
+        else if(submitString.equals("newMatcher"))
+        {
+            // If we have reached the matcher limit, do not allow any more matchers to be added.
+            if(editRule.getMatchers().size() >= Rule.MAX_MATCHERS)
+            {
+                bindingResult.rejectValue(
+                        "matchers",
+                        ValidationCodes.TOO_LONG,
+                        new Object[] {Rule.MAX_MATCHERS},
+                        "too many matchers specified");
+                return displayEditRuleForm(editRule, ruleId);
+            }
+            // Reload the page with the new matcher builder inside of the edit rule
+            return addNewMatcher(editRule, ruleId);
+        }
+        else
+        {
+            return saveRule(editRule, bindingResult, ruleId, submitString);
+        }
+    }
+    
+    private ModelAndView addNewMatcher(final EditExistingRule editRule,
```

Comment:
I do think you could eliminate much of this near-duplicate code via a common base class. We can discuss if you like.